File: SDKs/monarchinitiative.org/1.0.1/python/src/sdk/models/shared/searchresult.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class SearchResult:
    docs: Optional[list[dict[str, Any]]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('docs') }})
    facet_counts: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('facet_counts') }})
    highlighting: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('highlighting') }})
    num_found: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('numFound') }})
    

File: SDKs/monarchinitiative.org/1.0.1/typescript/src/sdk/models/shared/searchresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class SearchResult extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=docs" })
  docs?: Record<string, any>[];

  @SpeakeasyMetadata({ data: "json, name=facet_counts" })
  facetCounts?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=highlighting" })
  highlighting?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=numFound" })
  numFound?: number;
}
